File: src/main/java/org/snust/prefixsum/텐트세우기14719.java
package org.snust.prefixsum;

import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.ArrayList;
import java.util.List;
import java.util.StringTokenizer;

public class 텐트세우기14719 {
  public static void main(String[] args) throws IOException {
    BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
    StringTokenizer st = new StringTokenizer(br.readLine()," ");


    int H = Integer.parseInt(st.nextToken());
    int W = Integer.parseInt(st.nextToken());

    st = new StringTokenizer(br.readLine()," ");

    List<Integer> heightList = new ArrayList<>();
    for(int i = 0; i < W; i++) {
      heightList.add(Integer.parseInt(st.nextToken()));
    }

    int answer = 0;
    for (int i = 0; i < W-1; i++) {
      int leftMax = 0;
      for (int j = 0; j < i; j++) {
        leftMax = Math.max(leftMax, heightList.get(j));
      }

      int rightMax = 0;
      for (int j = i + 1; j < heightList.size(); j++) {
        rightMax = Math.max(rightMax, heightList.get(j));
      }

      int water = Math.min(leftMax, rightMax) - heightList.get(i);
      if (water > 0) {
        answer += water;
      }


    }
    System.out.println(answer);
  }
}

File: src/main/python/prefixsum/텐트세우기14719.py
H, W = map(int, input().split())
height_list = list(map(int, input().split()))
answer = 0

for i in range(1, W-1):  # 첫번째와 마지막 블록은 물이 고일 수 없음
  left_max = max(height_list[:i])  # 현재 위치의 왼쪽에서 가장 높은 블록
  right_max = max(height_list[i+1:])  # 현재 위치의 오른쪽에서 가장 높은 블록

  # 현재 위치에 고일 수 있는 물의 높이는 left_max와 right_max 중 작은 값에서 현재 높이를 뺀 값
  water = min(left_max, right_max) - height_list[i]
  if water > 0:
    answer += water

print(answer)
